Lib/sqlite3/_completer.py

```diff
@@ -0,0 +1,36 @@
+from _sqlite3 import SQLITE_KEYWORDS
+from contextlib import contextmanager
+
+_completion_matches = []
+
+
+def _complete(text, state):
+    global _completion_matches
+    if state == 0:
+        _completion_matches = [c + " " for c in SQLITE_KEYWORDS if c.startswith(text.upper())]
```

Comment:
```suggestion
    if state == 0:
        text_upper = text.upper()
        _completion_matches = [c for c in SQLITE_KEYWORDS if c.startswith(text_upper)]
```

Otherwise, I *think* `text.upper()` is going to be called for each `c`. Then you can do `return _completion_matches[state] + " "` instead (no need to store the extra whitespace) as it would create a new string while here you can just create a new reference to an existing string.